woocommerce/woocommerce-android
# Create survey product banner

Summary
==========
Fixes issue #2642, this PR delivers the following features:
- Add the `Give Feedback` and `Dismiss` buttons on the Product Banner
- Adapts the `FeedbackSurveyFragment` to support Products survey
- Creates the `FeedbackPrefs` to register all feedbacks actions the user may give
- Creates the `FeatureFeedbackSettings` as the main structure to register all user feedbacks requested with feature banners

Screenshots
==========
Light Mode
------------
| Click on the Product Banner to expand it | Click on the `Give Feedback` Button | Wait for the loading to finish and Interact with the survey | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![screencap-2020-09-04T141227 197Z](https://user-images.githubusercontent.com/5920403/92249397-62eb7e80-eea0-11ea-81ef-7f26c495e7da.png) | ![screencap-2020-09-04T141137 000Z](https://user-images.githubusercontent.com/5920403/92249395-6252e800-eea0-11ea-9eeb-d639f19ce4e0.png) | ![screencap-2020-09-04T141155 224Z](https://user-images.githubusercontent.com/5920403/92249392-61ba5180-eea0-11ea-8caf-18836e0043bc.png) | ![screencap-2020-09-04T141126 616Z](https://user-images.githubusercontent.com/5920403/92249385-5ff08e00-eea0-11ea-9c62-b9d089a7f213.png) |

Dark Mode
------------
| Click on the Product Banner to expand it | Click on the `Give Feedback` Button | Wait for the loading to finish and Interact with the survey | Verify if the expected view is presented and is fully functional | 
| ------------- | ------------- | ------------- | ------------- |
| ![screencap-2020-09-04T122521 617Z](https://user-images.githubusercontent.com/5920403/92249724-e2794d80-eea0-11ea-8809-aef242fb2c84.png) | ![screencap-2020-09-04T122524 916Z](https://user-images.githubusercontent.com/5920403/92249732-e4431100-eea0-11ea-9e3e-43479b85936a.png) | ![screencap-2020-09-04T141021 860Z](https://user-images.githubusercontent.com/5920403/92249737-e5743e00-eea0-11ea-9e8d-399d696cf711.png) | ![screencap-2020-09-04T141108 032Z](https://user-images.githubusercontent.com/5920403/92249738-e5743e00-eea0-11ea-9290-48a68892dc41.png) |

How to Test
==========
Scenario 1 - Give Feedback
------------
1. Access the `Products` section from the Woo app
2. Verify that the Product Banner is showing up and available and click in the `Give Feedback` button
3. Verify that the survey view is called, the loading dialog is visible until the survey is completely loaded
4. Interact with the survey until the conclusion, after finishing it, verify that the custom native Feedback Completed View is called instead of letting the WebView show the default web one.
5. Once inside the Feedback Completed View, verify that each button is working as expected and described
6. After returning from the Survey, verify that the Product Banner is no longer visible

Scenario 2 - Dismiss
------------
1. Access the `Products` section from the Woo app
2. Verify that the Product Banner is showing up and available and click in the `Dismiss` button
3. Verify that the Product Banner is no longer visible and doesn't show up after reopening the app or doing any normal interaction with it

Update release notes:
